pandas/tests/frame/test_indexing.py

```diff
@@ -2762,7 +2762,62 @@ def test_set_reset(self):
         result = df.reset_index()
         self.assertTrue(result['foo'].dtype, 'M8[ns, US/Eastern')
 
-        result = result.set_index('foo')
+        df = result.set_index('foo')
+        tm.assert_index_equal(df.index, idx)
+
+    def test_transpose(self):
+
+        result = self.df.T
+        expected = DataFrame(self.df.values.T)
+        expected.index = ['A', 'B']
+        assert_frame_equal(result, expected)
+
+
+class TestDataFrameIndexingUInt64(tm.TestCase, TestData):
+
+    _multiprocess_can_split_ = True
+
+    def setUp(self):
+        self.ir = Index(np.arange(3), dtype=np.uint64)
+        self.idx = Index([2**63, 2**63 + 5, 2**63 + 10], name='foo')
+
+        self.df = DataFrame({'A': self.idx, 'B': self.ir})
+
+    def test_setitem(self):
+
+        df = self.df
+        idx = self.idx
+
+        # setitem
+        df['C'] = idx
+        assert_series_equal(df['C'], Series(idx, name='C'))
+
+        df['D'] = 'foo'
+        df['D'] = idx
+        assert_series_equal(df['D'], Series(idx, name='D'))
+        del df['D']
+
+        # with nan
+        df2 = df.copy()
+        df2.iloc[1, 1] = pd.NaT
+        df2.iloc[1, 2] = pd.NaT
+        result = df2['B']
+        assert_series_equal(notnull(result), Series(
+            [True, False, True], name='B'))
+        assert_series_equal(df2.dtypes, Series([np.dtype('uint64'),
+                                                np.dtype('O'), np.dtype('O')],
+                                               index=['A', 'B', 'C']))
+
+    def test_set_reset(self):
+
+        idx = self.idx
+
+        # set/reset
+        df = DataFrame({'A': [0, 1, 2]}, index=idx)
+        result = df.reset_index()
+        self.assertEqual(result['foo'].dtype, np.dtype('uint64'))
```

Comment:
What do you mean?  I literally copied the test class right above.  This is testing `reset_index`.